lib/views/pages/profile/info_cards/profile_info_pop_up.dart

```diff
@@ -0,0 +1,74 @@
+import "package:flutter/material.dart";
+
+class ProfileInfoPopUp extends StatelessWidget {
+  static const popUpTitleKey = Key("profilePopUpTitle");
+  static const popUpDescriptionKey = Key("profilePopUpDescription");
+  static const popUpDeleteButtonKey = Key("profilePopUpDeleteButton");
+
+  const ProfileInfoPopUp({
+    super.key,
+    this.title,
+    required this.content,
+    required this.onDelete,
+  });
+
+  final String? title;
+  final String content;
+  final VoidCallback onDelete;
```

Comment:
The `onDelete` function is probably going to invoke some repository calls. 
Thus, this would mean that it returns a `Future<void>`.

So, I wonder if we shouldn't replace the type of `onDelete` as below:
```suggestion
  final Future<void> Function() onDelete;
```

And then we `await` it during the `onPressed` before navigating back.
What do you think ?

(If you choose to go with that, I believe you can apply this change in `profile_info_card` as well)

Reply:
Yes I am on it, this requires a few changes. I'll keep you posted.

Reply:
see PR comments